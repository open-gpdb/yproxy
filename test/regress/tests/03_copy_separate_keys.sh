#!/bin/bash
set -e

psql -h pg -U postgres -d test -c "INSERT INTO yezzey.yezzey_virtual_index VALUES ('/encrypted_file_old_key')"

echo 'Sample encrypted data, old key' | yp-client --config test/regress/conf/yproxy_old.yaml -l fatal put --encrypt 'encrypted_file_old_key'
yp-client --config test/regress/conf/yproxy_old.yaml -l fatal list ''
yp-client --config test/regress/conf/yproxy_old.yaml -l fatal cat --decrypt 'encrypted_file_old_key'

# Copy file encrypted with separate keys
yp-client copy --config test/regress/conf/yproxy.yaml --old-config test/regress/conf/yproxy_old.yaml --encrypt --decrypt --port 5432 --confirm --log-level error 'encrypted_file_old_key'
yp-client --config test/regress/conf/yproxy.yaml -l fatal list ''
yp-client --config test/regress/conf/yproxy.yaml -l fatal cat --decrypt 'encrypted_file_old_key'
