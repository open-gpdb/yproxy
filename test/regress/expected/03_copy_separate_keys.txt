INSERT 0 1
1
Object: {Name: "/encrypted_file_old_key", size: 481}
Sample encrypted data, old key
1
Object: {Name: "/encrypted_file_old_key", size: 481}
Sample encrypted data, old key
